mod/state-transition/pkg/core/state_processor_staking_test.go

```diff
@@ -0,0 +1,183 @@
+		*types.ExecutionPayload,
+		*types.ExecutionPayloadHeader,
+		engineprimitives.Withdrawals,
+	](t)
+	mocksSigner := &cryptomocks.BLSSigner{}
+
+	sp := core.NewStateProcessor[
+		*types.BeaconBlock,
+		*types.BeaconBlockBody,
+		*types.BeaconBlockHeader,
+		*TestBeaconStateT,
+		*transition.Context,
+		*types.Deposit,
+		*types.Eth1Data,
+		*types.ExecutionPayload,
+		*types.ExecutionPayloadHeader,
+		*types.Fork,
+		*types.ForkData,
+		*TestKVStoreT,
+		*types.Validator,
+		types.Validators,
+		*engineprimitives.Withdrawal,
+		engineprimitives.Withdrawals,
+		types.WithdrawalCredentials,
+	](
+		cs,
+		execEngine,
+		mocksSigner,
+	)
+
+	kvStore, err := initTestStore()
+	require.NoError(t, err)
+	beaconState := new(TestBeaconStateT).NewFromDB(kvStore, cs)
+
+	var (
+		maxBalance       = math.Gwei(cs.MaxEffectiveBalance())
+		minBalance       = math.Gwei(cs.EffectiveBalanceIncrement())
+		emptyAddress     = common.ExecutionAddress{}
+		emptyCredentials = types.NewCredentialsFromExecutionAddress(
+			emptyAddress,
+		)
+	)
+
+	// Setup initial state via genesis
+	// TODO: consider instead setting state artificially
+	var (
+		genDeposits = []*types.Deposit{
+			{
+				Pubkey:      [48]byte{0x01},
+				Credentials: emptyCredentials,
+				Amount:      maxBalance - 3*minBalance,
+				Index:       uint64(0),
+			},
+			{
+				Pubkey:      [48]byte{0x02},
+				Credentials: emptyCredentials,
+				Amount:      maxBalance - 6*minBalance,
+				Index:       uint64(1),
+			},
+		}
+		genPayloadHeader = new(types.ExecutionPayloadHeader).Empty()
+		genVersion       = version.FromUint32[common.Version](version.Deneb)
+	)
+
+	mocksSigner.On(
+		"VerifySignature",
+		mock.Anything, mock.Anything, mock.Anything,
+	).Return(nil)
+
+	_, err = sp.InitializePreminedBeaconStateFromEth1(
+		beaconState,
+		genDeposits,
+		genPayloadHeader,
+		genVersion,
+	)
+	require.NoError(t, err)
+
+	// create test inputs
+	var (
+		ctx = &transition.Context{
+			SkipPayloadVerification: true,
+			SkipValidateResult:      true,
+		}
+		blkDeposits = []*types.Deposit{
+			{
+				Pubkey:      genDeposits[0].Pubkey,
+				Credentials: emptyCredentials,
+				Amount:      minBalance, // avoid breaching maxBalance
+				Index:       genDeposits[0].Index,
+			},
+		}
+	)
+
+	// here we duly update state root, similarly to what we do in processSlot
+	genBlockHeader, err := beaconState.GetLatestBlockHeader()
+	require.NoError(t, err)
+	genStateRoot := beaconState.HashTreeRoot()
+	genBlockHeader.SetStateRoot(genStateRoot)
+
+	blk := &types.BeaconBlock{
+		Slot:          genBlockHeader.GetSlot() + 1,
+		ProposerIndex: genBlockHeader.GetProposerIndex(),
+		ParentRoot:    genBlockHeader.HashTreeRoot(),
+		StateRoot:     common.Root{},
+		Body: &types.BeaconBlockBody{
+			ExecutionPayload: &types.ExecutionPayload{
+				Timestamp:     10,
+				ExtraData:     []byte("testing"),
+				Transactions:  [][]byte{},
+				Withdrawals:   []*engineprimitives.Withdrawal{}, // no withdrawals
```

Comment:
this is the point of the test [here](https://github.com/berachain/beacon-kit/blob/41a76575535de5adb77811248bada0a2add084c0/mod/state-transition/pkg/core/state_processor_staking_test.go#L153), which is different from the base branch here: the list of withdrawals is empty instead of containing two withdrawals with zero amount.